sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestClassShouldHaveTestMethod_MSTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Tests.Diagnostics
+{
+    [TestClass]
+    class ClassTest2 // Noncompliant
+//        ^^^^^^^^^^
+    {
+    }
+
+    [TestClass]
+    class ClassTest4 // Noncompliant
+    {
+        public void Foo() { }
+    }
+
+    class ClassTest5
+    {
+        public void Foo() { }
+    }
+
+    class ClassTest6
+    {
+        public void Foo() { }
+    }
+
+    [TestClass]
+    class ClassTest9
+    {
+        [TestMethod]
+        public void Foo() { }
+    }
+
+    [TestClass]
+    class ClassTest12
+    {
+        [DataTestMethod]
+        [DataRow(1)]
+        public void Foo(int i) { }
+    }
+
+    [TestClass]
+    public abstract class MyCommonCode1
+    {
+    }
+
+    [TestClass]
+    public class MySubCommonCode1 : MyCommonCode1 // Noncompliant
```

Comment:
We could decide to ignore this case but for now I am forcing the base class to have tests.